components/Deck/ContentModulesPanel/DataSourcePanel/EditDataSource.js

```diff
@@ -7,10 +7,40 @@ import cancelEditDataSource from '../../../../actions/datasource/cancelEditDataS
 
 class EditDataSource extends React.Component {
     componentDidMount() {
+
+        $.fn.form.settings.rules.yearRule = (year) => {
+            let currDate = new Date();
+            let currYear = currDate.getYear() + 1900;;
+            let intYear = parseInt(year);
+            if (!intYear || intYear > currYear) {
+                return false;
+            }
+            return true;
+        };
         const dataSourceValidation = {
             fields: {
                 title: {
-                    identifier: 'title'
+                    identifier: 'title',
+                    rules: [{
+                        type: 'empty',
+                        prompt: 'This field cannot be empty.'
+                    }]
+                },
+                url: {
```

Comment:
The URL rule seems a bit strict, as it requires a protocol such as `http` etc. Not sure about usability concerns, since people might casually type in the URL with the protocol, e.g. `www.google.com`. This rule currently prevents this. Most people are not aware of what a proper URL is, or that it's required to include the `https://` in front of it. @abijames or @ali1k could comment on this as well.

Ideally, the form should be smart enough to do the check and insert it automatically, same way as browsers typically do. It would be safe to insert `http://` in such a case (instead of https) as most sites will do the redirect to https, whereas the ones that don't support https will not be able to handle that.

Reply:
In the original report, https://slidewiki.atlassian.net/browse/SWIK-1499 I also commented that the URL is not a required field in the case of Web Page / Web Documents. This is still the case, i.e. I can add a page with just a title and an emtpy URL and it will still be accepted. This needs to be done if this task is to be considered complete.

Reply:
Hmmm I get it, well actually the URL is not required, the only required field is the title, I didn't change that. Regarding the url validation.... hmmm yes, I didn't notice that the protocol was strictly required, but then are we going to assume that the url uses http or https always? Actually it makes some sense for me, but just want to clarify that.

Reply:
As I said, a site that supports `https` normally handles `http` requests as well by redirecting them to `https`. Sites that don't do that are very few and should reconsider, as it's trivial. On the other hand, a site that does not support `https` understandably cannot handle it and redirect to `http` for example. I think browsers handle that on their own, i.e. if the site does not respond to `https` they try switching to `http` but that's not always the case. 

So the safer approach would be to add a `http` protocol *when the user does not define it*, if possible with some smart library/coding.